container-search/src/main/java/com/yahoo/search/result/HitGroup.java

```diff
@@ -400,6 +401,19 @@ public ErrorHit getErrorHit() {
         getError(); // Make sure the error hit is updated
         return errorHit;
     }
+
+    /** 
+     * Removes the error hit of this.
+     * This removes all error messages of this and the query producing it.
+     *
+     * @return the error hit which was removed, or null if there were no errors
+     */
+    public ErrorHit removeErrorHit() {
+        updateHits(); // Consume and remove from the query producing this as well
+        ErrorHit removed = errorHit;
+        errorHit = null;
```

Comment:
shouldn't the error also be removed from the "hits" list?  if not, add a comment to clarify what the expected result is.

Reply:
You are entirely right ... this stuff is quite messy due to compatibility with the old model.
Now that that is long gone we can clean up though. I did most of that now, but added 2 todo's to avoid doing too much at once.